sql-to-dbsp-compiler/SQL-compiler/src/main/java/org/dbsp/sqlCompiler/compiler/frontend/ExpressionCompiler.java

```diff
@@ -977,12 +998,98 @@ node, new DBSPTypeBool(CalciteObject.EMPTY, false), DBSPOpcode.EQ,
                     arg0 = new DBSPApplyExpression("map", type, arg0.borrow(), cast);
                 }
 
-                String method = "array_append";
+                String method = getCallName(call);
                 if (arg0type.mayBeNull)
                     method += "N";
 
                 return new DBSPApplyExpression(node, method, type, arg0, arg1);
             }
+            case SORT_ARRAY: {
+                if (ops.size() == 1) {
+                    ops.add(1, new DBSPBoolLiteral(true));
+                }
+                DBSPExpression arg0 = ops.get(0);
+                DBSPExpression arg1 = ops.size() == 1 ? new DBSPBoolLiteral(true) : ops.get(1);
```

Comment:
this can no longer happen

Reply:
Oh yeah, I mixed two different approaches.